### PR TITLE
[SPARK-20894][SS] Resolve the checkpoint location in driver and use the resolved path in state store (branch-2.2)

### PR DESCRIPTION
**Pull request by zsxwing**
## What changes were proposed in this pull request?

Backport #18149 to 2.2.

## How was this patch tested?

Jenkins.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamSuite.scala [hunk 1]
```diff
@@ -614,6 +614,25 @@ class StreamSuite extends StreamTest {
     assertDescContainsQueryNameAnd(batch = 2)
     query.stop()
   }
+
+  test("should resolve the checkpoint path") {
```
Comment (zsxwing) [inline sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamSuite.scala]: Just test conflicts with 2.2.

**Comment by zsxwing** [comment]
cc @tdas

**Comment by markgrover** [comment]
Doesn't look like this made to 2.2. Should we mark the JIRA resolved with 2.3 as the target version. I am assuming it's too late for this to go to 2.2.0. May be retarget this for 2.2.1?

**Comment by zsxwing** [comment]
@markgrover I'm going to close this one. This is just an improvement and not need to go into 2.2.1.

**Comment by markgrover** [comment]
Sounds good to me. The JIRA was unresolved. I have resolved it with a fix version of 2.3.0

**Pull request closed without merging**